mongodbatlas/processes.go

```diff
@@ -0,0 +1,65 @@
+package mongodbatlas
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+)
+
+const processesPath = "groups/%s/processes"
+
+type ProcessesService interface {
+	List(context.Context, string, *ListOptions) ([]Process, *Response, error)
+}
+
+type ProcessesServiceOp struct {
+	Client RequestDoer
+}
+
+var _ ProcessesService = &ProcessesServiceOp{}
+
+type Process struct {
+	Created        string  `json:"created"`
+	GroupID        string  `json:"groupId"`
+	Hostname       string  `json:"hostname"`
+	ID             string  `json:"id"`
+	LastPing       string  `json:"lastPing"`
+	Links          []*Link `json:"links"`
+	Port           int     `json:"port"`
+	ShardName      string  `json:"shardName"`
+	ReplicaSetName string  `json:"replicaSetName"`
+	TypeName       string  `json:"typeName"`
+	Version        string  `json:"version"`
+}
+
+type processesResponse struct {
+	Links      []*Link   `json:"links,omitempty"`
+	Results    []Process `json:"results,omitempty"`
```

Comment:
[nit] We try to keep the results as pointers, this is also necessary for the `omitempty` in this case